service/gcs/main.go

```diff
@@ -139,27 +185,40 @@ func main() {
 		},
 	})
 	if err != nil {
-		logrus.WithError(err).Fatal("opengcs::main - failed to create containers cgroup")
+		logrus.WithError(err).Fatal("failed to create containers cgroup")
 	}
 	defer containersControl.Delete()
-	gcsLimit := int64(*gcsMemLimitBytes)
-	gcsControl, err := cgroups.New(cgroups.V1, cgroups.StaticPath("/gcs"), &oci.LinuxResources{
-		Memory: &oci.LinuxMemory{
-			Limit: &gcsLimit,
-		},
-	})
+
+	gcsControl, err := cgroups.New(cgroups.V1, cgroups.StaticPath("/gcs"), &oci.LinuxResources{})
 	if err != nil {
-		logrus.WithError(err).Fatal("opengcs::main - failed to create gcs cgroup")
+		logrus.WithError(err).Fatal("failed to create gcs cgroup")
 	}
 	defer gcsControl.Delete()
 	if err := gcsControl.Add(cgroups.Process{Pid: os.Getpid()}); err != nil {
-		logrus.WithError(err).Fatal("opengcs::main - failed add gcs pid to gcs cgroup")
+		logrus.WithError(err).Fatal("failed add gcs pid to gcs cgroup")
+	}
+
+	event := cgroups.MemoryThresholdEvent(*gcsMemLimitBytes, false)
+	gefd, err := gcsControl.RegisterMemoryEvent(event)
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to register memory threshold for gcs cgroup")
+	}
+	gefdFile := os.NewFile(gefd, "gefd")
+	defer gefdFile.Close()
+
+	oom, err := containersControl.OOMEventFD()
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to retrieve the container cgroups oom eventfd")
 	}
+	oomFile := os.NewFile(oom, "cefd")
+	defer oomFile.Close()
 
+	go readMemoryEvents(gefdFile, "gcs", int64(*gcsMemLimitBytes), gcsControl)
```

Comment:
I think we should make these cgroup names be the exact value, so `"/gcs"` instead of `"gcs"`, and `"/containers"` instead of `"containers"`.